staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/config/client.go

```diff
@@ -111,7 +111,9 @@ func (cm *ClientManager) HookClient(h *v1beta1.Webhook) (*rest.RESTClient, error
 	}
 
 	complete := func(cfg *rest.Config) (*rest.RESTClient, error) {
-		cfg.TLSClientConfig.CAData = h.ClientConfig.CABundle
+		if len(h.ClientConfig.CABundle) != 0 {
```

Comment:
The client manager has special `kubernetes.default.svc` handling via the wrapper mechanism of the webhook, set up in kube-apiserver to return the loopback client. That client is (very) special, i.e. it has its own certificate and is selected with a special (fake) SNI ServerName. So overriding those values break the loopback client.

Reply:
Here is the code: https://github.com/sttts/kubernetes/blob/69e2a9cb48fd49cf1449042984ea9c25edc0598c/cmd/kube-apiserver/app/server.go#L454

Reply:
Yeah, I saw that code, and I agree it's a valid case, so I suggested that we mark the `CABundle` field as optional instead of [required](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/admissionregistration/v1beta1/types.go#L264).

Reply:
The same webhook registration must work for kube-apiserver (potentially via loopback) or by user-provider aggregated apiservers (potentially via in-cluster config). I guess in both cases the `CABundle` is not necessary. So yes, optional might make sense.